src/App/Platforms/Android/Tiles/AutofillTileService.cs

```diff
@@ -76,7 +77,24 @@ private void ScanAndFill()
             var intent = new Intent(this, typeof(AccessibilityActivity));
             intent.SetFlags(ActivityFlags.NewTask | ActivityFlags.SingleTop | ActivityFlags.ClearTop);
             intent.PutExtra("autofillTileClicked", true);
-            StartActivityAndCollapse(intent);
+
+            //For Android 14+ We need to use PendingIntent instead of Intent directly. Older versions still need to use Intent.
+            if (Build.VERSION.SdkInt >= BuildVersionCodes.UpsideDownCake)
+            {
+                var pendingIntent = PendingIntent.GetActivity(
+                    ApplicationContext,
+                    0,
+                    intent,
+                    PendingIntentFlags.Mutable | PendingIntentFlags.UpdateCurrent
+                );
+                if (pendingIntent == null) { return; }
+
+                StartActivityAndCollapse(pendingIntent);
+            }
+            else
+            {
+                StartActivityAndCollapse(intent);
+            }
```

Comment:
🎨 Add extension method from `Intent` to reuse this code between the three services which would have the Android 14 check inside so the code is cleaner. Something like `intent.ToActivityPendingIntentIfNecessary()` or anything else more descriptive.

Reply:
@fedemkr The return type is different and it's specific for TileService but it can still be shared for sure.